test/models/host_test.rb

```diff
@@ -3122,6 +3122,27 @@ def to_managed!
     end
   end
 
+  context "recreating host TFTP config" do
+    setup do
+      @nic = FactoryGirl.build(:nic_primary_and_provision)
+      @nic.expects(:rebuild_tftp).returns(true)
+      @nic.expects(:rebuild_dns).never
+      @nic.expects(:rebuild_dhcp).never
+      Nic::Managed.expects(:rebuild_methods).returns(:rebuild_dhcp => "DHCP", :rebuild_dns => "DNS", :rebuild_tftp => "TFTP")
+    end
+
+    test "recreate TFTP config with success" do
+      Host::Managed.expects(:rebuild_methods).returns(:rebuild_test => "TEST")
+      host = FactoryGirl.build(:host, :interfaces => [@nic])
+      host.expects(:rebuild_test).never
+      result = host.recreate_config(['TFTP'])
+      refute result["DHCP"]
+      refute result["DNS"]
+      assert result["TFTP"]
+      refute result["TEST"]
+    end
+  end
```

Comment:
Can you also add edge case tests? When only is empty array and when it's filled with all possible values (you can dynamically generate them).